### PR TITLE
 IN-141: Added recommended K8s labels to Curator.

### PR DESCRIPTION
**Pull request by michaelamiethyst**

### CODE REVIEW

File: charts/curator/templates/configmap.yaml [hunk 1]
```diff
@@ -2,6 +2,14 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: curator-config
+  labels:
+    app.kubernetes.io/name: {{ template "name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    app.kubernetes.io/component: infrastructure
+    app.kubernetes.io/part-of: logging-central
```
Comment (mikeln) [inline charts/curator/templates/configmap.yaml]: Don't like this fixed value.   But I guess there is no .Chart. value that is usable.

File: charts/curator/templates/configmap.yaml [hunk 2]
```diff
@@ -2,6 +2,14 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: curator-config
+  labels:
+    app.kubernetes.io/name: {{ template "name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (DStorck) [inline charts/curator/templates/configmap.yaml]: i don't think we currently have this field in our chart

File: charts/curator/templates/cronjob.yaml [hunk 3]
```diff
@@ -2,6 +2,14 @@ apiVersion: {{ template "cronjob.apiVersion" . }}
 kind: CronJob
```
Comment (mikeln) [inline charts/curator/templates/cronjob.yaml]: The apiVersion for this file should be changed to the current correct one.  Not the select version selector in _helpers.tpl.
Also the version selector in _helpers.tpl needs to be removed.
(All this is depends on if the current way to do this is finally a real version, and we don't need to support different apiVersions based on the Kubernetes cluster version we are supporting.).  Should be essentially the latest kubernetes version, and whatever the apiVersion for "job" is in that version.

Comment (DStorck) [reply charts/curator/templates/cronjob.yaml]: yes that version check can be taken out or updated , it was back when cronjobs were transitioning from beta

File: charts/curator/templates/secret.yaml [hunk 4]
```diff
@@ -7,5 +7,12 @@ metadata:
     app: {{ template "name" . }}
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
+    app.kubernetes.io/name: {{ template "name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (DStorck) [inline charts/curator/templates/secret.yaml]: same here. so we can take this out, or add this field to Chart.yaml

File: charts/curator/templates/configmap.yaml (hunk 1 above)
Comment (michaelamiethyst) [reply charts/curator/templates/configmap.yaml]: The helm-docs recommended a hard-coded value for this field at https://docs.helm.sh/chart_best_practices/#labels-and-annotations

File: charts/curator/templates/configmap.yaml (hunk 2 above)
Comment (michaelamiethyst) [reply charts/curator/templates/configmap.yaml]: I changed it to .Chart.version

Comment (michaelamiethyst) [reply charts/curator/templates/configmap.yaml]: Make that I kept it at .Chart.AppVersion and added AppVersion to our chart.yaml.

File: charts/curator/templates/cronjob.yaml (hunk 3 above)
Comment (michaelamiethyst) [reply charts/curator/templates/cronjob.yaml]: I took it out of _helpers.tpl and hard-codeded it to the current batch/v1beta1.

File: charts/curator/templates/secret.yaml (hunk 4 above)
Comment (michaelamiethyst) [reply charts/curator/templates/secret.yaml]: Added.

**Comment by michaelamiethyst** [comment]
I've fixed all suggested changes.  Let me know if it looks good now.

File: charts/curator/Chart.yaml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 name: curator
 version: 0.1.0
+AppVersion: 0.1.0
```
Comment (mattfarina) [inline charts/curator/Chart.yaml]: The `appVersion` should be curators version which I believe is 5.6.0.

Also notice the a is lowercase at the start.

Comment (DStorck) [inline charts/curator/Chart.yaml]: this field should reflect the version of the app ( curator ) which we are deploying with this chart - currently we deploy curator `5.6.0`

File: charts/curator/templates/cronjob.yaml [hunk 6]
```diff
@@ -1,7 +1,15 @@
-apiVersion: {{ template "cronjob.apiVersion" . }}
+apiVersion: batch/v1beta1
```
Comment (mikeln) [inline charts/curator/templates/cronjob.yaml]: is this the most current apiVersion that has cronjob?   I suspect it may be out of beta...maybe?

Comment (mikeln) [reply charts/curator/templates/cronjob.yaml]: ok, looks like cronjob is almost the only thing left in v1beta1  (as of kube v1.13)

File: charts/curator/Chart.yaml (hunk 5 above)
Comment (michaelamiethyst) [reply charts/curator/Chart.yaml]: Thank you for the info.  Fixed.

Comment (michaelamiethyst) [reply charts/curator/Chart.yaml]: Thank you.  Fixed.

**Comment by michaelamiethyst** [comment]
I've made all requested changes.  Thank you for your time.

**Pull request merged**